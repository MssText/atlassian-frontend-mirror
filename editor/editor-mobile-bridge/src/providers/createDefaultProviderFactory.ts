import { ProviderFactory } from '@atlaskit/editor-common';
import { createEditorProviders } from './createEditorProviders';
import mockEmojiProvider from './mockEmojiProvider';

export function createDefaultProviderFactory() {
  const {
    mediaProvider,
    mentionProvider,
    cardProvider,
    taskDecisionProvider,
  } = createEditorProviders();

  return ProviderFactory.create({
    mediaProvider,
    mentionProvider,
    cardProvider,
    emojiProvider: Promise.resolve(mockEmojiProvider),
    taskDecisionProvider,
  });
}
