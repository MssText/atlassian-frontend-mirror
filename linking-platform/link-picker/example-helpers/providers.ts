import fetchMock from 'fetch-mock/cjs/client';
import mockRecentData from './mock-data';
import MockFlightDeckManifest from './flight-deck/_manifest.json';
import MockFlightDeckResponse from './flight-deck/mockFDResponse.json';

const activityUrlRegex = /\/gateway\/api\/graphql/;
const fdUrlRegex = /\/recent-work-service/;
const fdManifestRegex = /\/recent-work-service\/_manifest\.json/;

export const mockRecentClient = (delay?: number) => {
  fetchMock.post(
    activityUrlRegex,
    (_: string, options: { body: string }) => mockRecentData.ACTIVITIES_DATA,
    { method: 'POST', overwriteRoutes: true, delay },
  );

  fetchMock.get(
    fdManifestRegex,
    (_: string, options: { body: string }) => MockFlightDeckManifest,
    { method: 'GET', overwriteRoutes: true, delay },
  );
  fetchMock.get(
    fdUrlRegex,
    (_: string, options: { body: string }) => MockFlightDeckResponse,
    { method: 'GET', overwriteRoutes: true, delay },
  );
};
