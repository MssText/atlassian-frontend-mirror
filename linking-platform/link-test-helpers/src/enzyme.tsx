import { Component } from 'react';

import { mount, ReactWrapper, shallow, ShallowWrapper } from 'enzyme';
import { IntlProvider } from 'react-intl-next';

export function mountWithIntl<
  P = {},
  S = {},
  C extends Component<P, S> = Component<P, S>
>(jsx: JSX.Element): ReactWrapper<P, S, C> {
  return mount<C, P, S>(jsx, {
    wrappingComponent: IntlProvider,
    wrappingComponentProps: { locale: 'en' },
  } as any);
}

export function shallowWithIntl<
  P = {},
  S = {},
  C extends Component<P, S> = Component<P, S>
>(jsx: JSX.Element): ShallowWrapper<P, S, C> {
  return shallow<C, P, S>(jsx, {
    wrappingComponent: IntlProvider,
    wrappingComponentProps: { locale: 'en' },
  } as any);
}
