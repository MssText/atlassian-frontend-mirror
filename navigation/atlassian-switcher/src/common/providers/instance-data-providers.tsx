import React from 'react';

import { fetchJson, postJson } from '../utils/fetch';
import asDataProvider, { ResultComplete, Status } from './as-data-provider';
import {
  CollaborationGraphContainersResponse,
  Permissions,
  RecentContainersResponse,
  UserPermissionResponse,
  WithCloudId,
  XFlowSettingsResponse,
  ProviderResults,
} from '../../types';
import { withCached } from '../utils/with-cached';
import withHandleOptionalCloudId from './with-handle-optional-cloud-id';

// Recent activity api
const fetchRecentContainers = ({ cloudId }: WithCloudId) =>
  fetchJson<RecentContainersResponse>(
    `/gateway/api/activity/api/client/recent/containers?cloudId=${cloudId}`,
  );

const fetchCollaborationGraphRecentContainers = ({ cloudId }: WithCloudId) =>
  postJson<CollaborationGraphContainersResponse>(
    'gateway/api/v1/collaborationgraph/user/container',
    {
      containerTypes: ['confluenceSpace', 'jiraProject'],
      context: {
        contextType: 'atlassianSwitcher',
        principalId: 'context',
        siteId: cloudId,
      },
      expanded: true,
      maxNumberOfResults: 6,
      userId: 'context',
    },
  );

const RealRecentContainersProvider = asDataProvider(
  'recentContainers',
  fetchRecentContainers,
);

const RealCollaborationGraphRecentContainersProvider = asDataProvider(
  'collaborationGraphRecentContainers',
  fetchCollaborationGraphRecentContainers,
);

export const emptyRecentContainers: ResultComplete<RecentContainersResponse> = {
  status: Status.COMPLETE,
  data: { data: [] },
};

export const emptyCollaborationGraphRecentContainers: ResultComplete<CollaborationGraphContainersResponse> = {
  status: Status.COMPLETE,
  data: { collaborationGraphEntities: [] },
};

export const nullRecentContainers: ResultComplete<RecentContainersResponse> = {
  status: Status.COMPLETE,
  data: { data: null },
};

export const nullCollaborationGraphRecentContainers: ResultComplete<CollaborationGraphContainersResponse> = {
  status: Status.COMPLETE,
  data: { collaborationGraphEntities: null },
};

export const CollaborationGraphRecentContainersProvider = withHandleOptionalCloudId(
  ({
    cloudId,
    disableRecentContainers,
    enableCollaborationGraphRecentContainers,
    children,
  }: {
    cloudId: string;
    disableRecentContainers?: boolean;
    children: (
      collaborationGraphRecentContainers: ProviderResults['collaborationGraphRecentContainers'],
    ) => React.ReactNode;
    enableCollaborationGraphRecentContainers?: boolean;
  }) => {
    if (disableRecentContainers) {
      return (
        <React.Fragment>
          {children(emptyCollaborationGraphRecentContainers)}
        </React.Fragment>
      );
    }

    if (enableCollaborationGraphRecentContainers) {
      return (
        <RealCollaborationGraphRecentContainersProvider cloudId={cloudId}>
          {children}
        </RealCollaborationGraphRecentContainersProvider>
      );
    }

    // Returns null so that map-results-to-switcher-props.ts knows this provider is off
    return (
      <React.Fragment>
        {children(nullCollaborationGraphRecentContainers)}
      </React.Fragment>
    );
  },
  emptyCollaborationGraphRecentContainers.data,
);

export const RecentContainersProvider = withHandleOptionalCloudId(
  ({
    cloudId,
    disableRecentContainers,
    enableCollaborationGraphRecentContainers,
    children,
  }: {
    disableRecentContainers?: boolean;
    children: (
      recentContainers: ProviderResults['recentContainers'],
    ) => React.ReactNode;
    enableCollaborationGraphRecentContainers?: boolean;
  } & WithCloudId) => {
    if (disableRecentContainers) {
      return <React.Fragment>{children(emptyRecentContainers)}</React.Fragment>;
    }

    if (!enableCollaborationGraphRecentContainers) {
      return (
        <RealRecentContainersProvider cloudId={cloudId}>
          {children}
        </RealRecentContainersProvider>
      );
    }

    return <React.Fragment>{children(nullRecentContainers)}</React.Fragment>;
  },
  emptyRecentContainers.data,
);

// Permissions api
type FetchPermissionParamsType = WithCloudId & {
  permissionId: Permissions;
};
const fetchPermission = withCached(
  ({ cloudId, permissionId }: FetchPermissionParamsType) =>
    postJson<UserPermissionResponse>(`/gateway/api/permissions/permitted`, {
      permissionId,
      resourceId: `ari:cloud:platform::site/${cloudId}`,
    }).then(permission => permission.permitted),
);

export const UserPermissionProvider = withHandleOptionalCloudId(
  asDataProvider('permission', fetchPermission, fetchPermission.cached),
  false,
);

// Xflow settings api
const fetchXflowSettings = withCached(({ cloudId }: WithCloudId) =>
  fetchJson<XFlowSettingsResponse>(
    `/gateway/api/site/${cloudId}/setting/xflow`,
  ).then(xFlowSettings =>
    xFlowSettings['product-suggestions-enabled'] !== undefined
      ? xFlowSettings['product-suggestions-enabled']
      : true,
  ),
);

export const XFlowSettingsProvider = withHandleOptionalCloudId(
  asDataProvider(
    'xflowSettings',
    fetchXflowSettings,
    fetchXflowSettings.cached,
  ),
  true,
);

export const prefetchAll = ({ cloudId }: WithCloudId) => {
  fetchXflowSettings({ cloudId });
  fetchPermission({
    cloudId,
    permissionId: Permissions.ADD_PRODUCTS,
  });
  fetchPermission({ cloudId, permissionId: Permissions.MANAGE });
};

export const resetAll = () => {
  fetchXflowSettings.reset();
  fetchPermission.reset();
};
