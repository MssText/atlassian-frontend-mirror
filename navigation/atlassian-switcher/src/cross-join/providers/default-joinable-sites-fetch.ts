import { JoinableSitesResponse, ProductKey } from '../../types';
import { fetchJson } from '../../common/utils/fetch';

const joinSupportedProducts: ProductKey[] = [
  ProductKey.JIRA_SOFTWARE,
  ProductKey.JIRA_SERVICE_DESK,
  ProductKey.JIRA_CORE,
  ProductKey.CONFLUENCE,
];

export const fetchJoinableSites = (
  products: ProductKey[],
  baseUrl: string = '',
): Promise<JoinableSitesResponse> => {
  return fetchJson<JoinableSitesResponse>(
    `${baseUrl}/trello-cross-product-join/recommended-sites`,
    {
      method: 'post',
      body: JSON.stringify({ products }),
      headers: {
        'Content-Type': 'application/json',
      },
    },
  );
};

export const defaultFetchData = (baseUrl: string) => () =>
  fetchJoinableSites(joinSupportedProducts, baseUrl);
