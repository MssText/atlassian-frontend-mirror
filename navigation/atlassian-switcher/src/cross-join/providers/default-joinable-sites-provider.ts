import {
  createProviderWithCustomFetchData,
  ExportedDataProvider,
} from '../../common/providers/create-data-provider';
import { JoinableSitesResponse } from '../../types';

export type JoinableSiteDataFetcher = () => Promise<JoinableSitesResponse>;

export const fetchEmptyData: JoinableSiteDataFetcher = () =>
  Promise.resolve({ sites: [] });

export const createJoinableSitesProvider = (
  fetchData: JoinableSiteDataFetcher = fetchEmptyData,
): ExportedDataProvider<JoinableSitesResponse> => {
  const wrappedFetchData = withTryCatch(fetchData, fetchEmptyData);
  return createProviderWithCustomFetchData<JoinableSitesResponse>(
    'joinableSites',
    wrappedFetchData,
  );
};

export const withTryCatch = (fn: Function, fallback: Function) => () => {
  try {
    return fn();
  } catch (e) {
    return fallback();
  }
};
