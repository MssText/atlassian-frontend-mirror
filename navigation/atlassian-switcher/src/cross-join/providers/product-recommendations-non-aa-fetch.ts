import { JoinableSitesResponse, ProductKey } from '../../types';
import { fetchJson } from '../../common/utils/fetch';

const joinSupportedProducts: ProductKey[] = [
  ProductKey.JIRA_SOFTWARE,
  ProductKey.JIRA_SERVICE_DESK,
  ProductKey.JIRA_CORE,
  ProductKey.CONFLUENCE,
];

export const fetchJoinableSitesRecommended = (
  products: ProductKey[],
  baseUrl: string = '',
): Promise<JoinableSitesResponse> => {
  return fetchJson<JoinableSitesResponse>(
    `${baseUrl}/trello-cross-product-recommended/recommended-sites`,
    {
      method: 'post',
      body: JSON.stringify({ products }),
      headers: {
        'Content-Type': 'application/json',
      },
    },
  );
};

export const productRecommendationsNonAAFetchData = (baseUrl: string) => () =>
  fetchJoinableSitesRecommended(joinSupportedProducts, baseUrl);
