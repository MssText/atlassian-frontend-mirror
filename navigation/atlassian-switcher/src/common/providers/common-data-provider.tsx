import React from 'react';

import { ProviderResults } from '../../types';

import {
  UserPermissionProvider,
  XFlowSettingsProvider,
  CollaborationGraphRecentContainersProvider,
} from './instance-data-providers';
import { RecommendationsEngineProvider } from '../../cross-flow/providers/recommendations-provider';
import { Permissions, RecommendationsFeatureFlags } from '../../types';

interface CommonDataProviderProps {
  cloudId?: string;
  enableRecentContainers: boolean;
  recommendationsFeatureFlags?: RecommendationsFeatureFlags;
  children: (props: {
    managePermission: ProviderResults['managePermission'];
    addProductsPermission: ProviderResults['addProductsPermission'];
    isXFlowEnabled: ProviderResults['isXFlowEnabled'];
    productRecommendations: ProviderResults['productRecommendations'];
    collaborationGraphRecentContainers: ProviderResults['collaborationGraphRecentContainers'];
  }) => React.ReactElement<any>;
}

export default ({
  cloudId,
  children,
  recommendationsFeatureFlags,
  enableRecentContainers,
}: CommonDataProviderProps) => {
  return (
    <CollaborationGraphRecentContainersProvider
      cloudId={cloudId}
      enableRecentContainers={enableRecentContainers}
    >
      {collaborationGraphRecentContainers => (
        <UserPermissionProvider
          cloudId={cloudId}
          permissionId={Permissions.MANAGE}
        >
          {managePermission => (
            <UserPermissionProvider
              cloudId={cloudId}
              permissionId={Permissions.ADD_PRODUCTS}
            >
              {addProductsPermission => (
                <XFlowSettingsProvider cloudId={cloudId}>
                  {isXFlowEnabled => (
                    <RecommendationsEngineProvider
                      featureFlags={recommendationsFeatureFlags}
                    >
                      {productRecommendations =>
                        children({
                          managePermission,
                          addProductsPermission,
                          isXFlowEnabled,
                          productRecommendations,
                          collaborationGraphRecentContainers,
                        })
                      }
                    </RecommendationsEngineProvider>
                  )}
                </XFlowSettingsProvider>
              )}
            </UserPermissionProvider>
          )}
        </UserPermissionProvider>
      )}
    </CollaborationGraphRecentContainersProvider>
  );
};
