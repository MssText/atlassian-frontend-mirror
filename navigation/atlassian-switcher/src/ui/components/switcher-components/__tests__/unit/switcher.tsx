import React from 'react';
import { shallow } from 'enzyme';
import Switcher from '../../switcher';
import { Status } from '../../../../../common/providers/as-data-provider';
import {
  FeatureMap,
  SyntheticProviderResults,
  ProviderResults,
} from '../../../../../types';

const noop = () => void 0;
const providerResults: ProviderResults & SyntheticProviderResults = {
  joinableSites: {
    data: {
      sites: [],
    },
    status: Status.COMPLETE,
  },
  isXFlowEnabled: {
    data: false,
    status: Status.COMPLETE,
  },
  provisionedProducts: {
    data: {},
    status: Status.COMPLETE,
  },
  addProductsPermission: {
    data: false,
    status: Status.COMPLETE,
  },
  managePermission: {
    data: false,
    status: Status.COMPLETE,
  },
  recentContainers: {
    data: { data: [] },
    status: Status.COMPLETE,
  },
  collaborationGraphRecentContainers: {
    data: { collaborationGraphEntities: [] },
    status: Status.COMPLETE,
  },
  customLinks: {
    data: [],
    status: Status.COMPLETE,
  },
  userSiteData: {
    data: { currentSite: { url: '', products: [] } },
    status: Status.COMPLETE,
  },
  availableProducts: {
    data: { sites: [] },
    status: Status.COMPLETE,
  },
  productRecommendations: {
    data: [],
    status: Status.COMPLETE,
  },
};

describe('Switcher', () => {
  it('should render sections with headers by default', () => {
    expect(
      shallow(
        <Switcher
          triggerXFlow={noop}
          hasLoaded
          hasLoadedCritical
          hasLoadedInstanceProviders
          onDiscoverMoreClicked={noop}
          licensedProductLinks={[]}
          suggestedProductLinks={[]}
          joinableSiteLinks={[]}
          fixedLinks={[]}
          adminLinks={[]}
          recentLinks={[]}
          customLinks={[]}
          discoverSectionLinks={[]}
          rawProviderResults={providerResults}
          features={{} as FeatureMap}
        />,
      ),
    ).toMatchSnapshot();
  });

  it('should be able to disable switch to section header with "disableSwitchToHeading"', () => {
    expect(
      shallow(
        <Switcher
          triggerXFlow={noop}
          hasLoaded
          hasLoadedCritical
          hasLoadedInstanceProviders
          onDiscoverMoreClicked={noop}
          licensedProductLinks={[]}
          suggestedProductLinks={[]}
          joinableSiteLinks={[]}
          fixedLinks={[]}
          adminLinks={[]}
          recentLinks={[]}
          customLinks={[]}
          disableSwitchToHeading
          discoverSectionLinks={[]}
          rawProviderResults={providerResults}
          features={{} as FeatureMap}
        />,
      ),
    ).toMatchSnapshot();
  });
});
