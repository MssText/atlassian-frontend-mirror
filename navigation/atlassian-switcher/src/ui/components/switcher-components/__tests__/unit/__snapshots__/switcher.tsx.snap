// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switcher should be able to disable switch to section header with "disableSwitchToHeading" 1`] = `
<NavigationAnalyticsContext
  data={
    Object {
      "attributes": Object {
        "itemsCount": 0,
      },
    }
  }
>
  <SwitcherWrapper
    appearance="drawer"
  >
    <WithAnalyticsEvents(_class3)
      data={
        Object {
          "adminLinks": Array [],
          "fixedLinks": Array [],
          "joinableSiteLinks": Array [],
          "joinableSiteProductLinks": Array [],
          "licensedProducts": Array [],
          "numberOfSites": 0,
          "suggestedProducts": Array [],
        }
      }
      subject="atlassianSwitcher"
    />
    <WithAnalyticsEvents(_class2)
      data={
        Object {
          "duration": 0,
        }
      }
      subject="atlassianSwitcherAvailableProducts"
    />
    <WithAnalyticsEvents(_class)
      data={
        Object {
          "duration": 0,
          "emptyRender": true,
        }
      }
      subject="atlassianSwitcherJoinableSites"
    />
    <WithAnalyticsEvents(_class)
      data={
        Object {
          "duration": 0,
          "emptyRender": true,
        }
      }
      subject="atlassianSwitcherRecommendedProducts"
    />
    <WithAnalyticsEvents(_class)
      data={
        Object {
          "duration": 0,
          "emptyRender": true,
        }
      }
      subject="atlassianSwitcherDiscoverMore"
    />
    <WithAnalyticsEvents(_class)
      data={
        Object {
          "duration": 0,
          "emptyRender": true,
        }
      }
      subject="atlassianSwitcherRecentContainers"
    />
    <WithAnalyticsEvents(_class)
      data={
        Object {
          "duration": 0,
          "emptyRender": true,
        }
      }
      subject="atlassianSwitcherCustomLinks"
    />
    <SwitchToSection
      adminLinks={Array []}
      appearance="drawer"
      disableHeading={true}
      fixedLinks={Array []}
      licensedProductLinks={Array []}
      onDiscoverMoreClicked={[Function]}
      suggestedProductLinks={Array []}
      triggerXFlow={[Function]}
    />
    <CrossJoinSection
      appearance="drawer"
      joinableSiteLinks={Array []}
      licensedProductLinks={Array []}
      rawProviderResults={
        Object {
          "addProductsPermission": Object {
            "data": false,
            "status": "complete",
          },
          "availableProducts": Object {
            "data": Object {
              "sites": Array [],
            },
            "status": "complete",
          },
          "collaborationGraphRecentContainers": Object {
            "data": Object {
              "collaborationGraphEntities": Array [],
            },
            "status": "complete",
          },
          "customLinks": Object {
            "data": Array [],
            "status": "complete",
          },
          "isXFlowEnabled": Object {
            "data": false,
            "status": "complete",
          },
          "joinableSites": Object {
            "data": Object {
              "sites": Array [],
            },
            "status": "complete",
          },
          "managePermission": Object {
            "data": false,
            "status": "complete",
          },
          "productRecommendations": Object {
            "data": Array [],
            "status": "complete",
          },
          "provisionedProducts": Object {
            "data": Object {},
            "status": "complete",
          },
          "recentContainers": Object {
            "data": Object {
              "data": Array [],
            },
            "status": "complete",
          },
          "userSiteData": Object {
            "data": Object {
              "currentSite": Object {
                "products": Array [],
                "url": "",
              },
            },
            "status": "complete",
          },
        }
      }
    />
    <RecentSection
      appearance="drawer"
      recentLinks={Array []}
    />
    <CustomLinksSection
      appearance="drawer"
      customLinks={Array []}
    />
  </SwitcherWrapper>
</NavigationAnalyticsContext>
`;

exports[`Switcher should render sections with headers by default 1`] = `
<NavigationAnalyticsContext
  data={
    Object {
      "attributes": Object {
        "itemsCount": 0,
      },
    }
  }
>
  <SwitcherWrapper
    appearance="drawer"
  >
    <WithAnalyticsEvents(_class3)
      data={
        Object {
          "adminLinks": Array [],
          "fixedLinks": Array [],
          "joinableSiteLinks": Array [],
          "joinableSiteProductLinks": Array [],
          "licensedProducts": Array [],
          "numberOfSites": 0,
          "suggestedProducts": Array [],
        }
      }
      subject="atlassianSwitcher"
    />
    <WithAnalyticsEvents(_class2)
      data={
        Object {
          "duration": 0,
        }
      }
      subject="atlassianSwitcherAvailableProducts"
    />
    <WithAnalyticsEvents(_class)
      data={
        Object {
          "duration": 0,
          "emptyRender": true,
        }
      }
      subject="atlassianSwitcherJoinableSites"
    />
    <WithAnalyticsEvents(_class)
      data={
        Object {
          "duration": 0,
          "emptyRender": true,
        }
      }
      subject="atlassianSwitcherRecommendedProducts"
    />
    <WithAnalyticsEvents(_class)
      data={
        Object {
          "duration": 0,
          "emptyRender": true,
        }
      }
      subject="atlassianSwitcherDiscoverMore"
    />
    <WithAnalyticsEvents(_class)
      data={
        Object {
          "duration": 0,
          "emptyRender": true,
        }
      }
      subject="atlassianSwitcherRecentContainers"
    />
    <WithAnalyticsEvents(_class)
      data={
        Object {
          "duration": 0,
          "emptyRender": true,
        }
      }
      subject="atlassianSwitcherCustomLinks"
    />
    <SwitchToSection
      adminLinks={Array []}
      appearance="drawer"
      fixedLinks={Array []}
      licensedProductLinks={Array []}
      onDiscoverMoreClicked={[Function]}
      suggestedProductLinks={Array []}
      triggerXFlow={[Function]}
    />
    <CrossJoinSection
      appearance="drawer"
      joinableSiteLinks={Array []}
      licensedProductLinks={Array []}
      rawProviderResults={
        Object {
          "addProductsPermission": Object {
            "data": false,
            "status": "complete",
          },
          "availableProducts": Object {
            "data": Object {
              "sites": Array [],
            },
            "status": "complete",
          },
          "collaborationGraphRecentContainers": Object {
            "data": Object {
              "collaborationGraphEntities": Array [],
            },
            "status": "complete",
          },
          "customLinks": Object {
            "data": Array [],
            "status": "complete",
          },
          "isXFlowEnabled": Object {
            "data": false,
            "status": "complete",
          },
          "joinableSites": Object {
            "data": Object {
              "sites": Array [],
            },
            "status": "complete",
          },
          "managePermission": Object {
            "data": false,
            "status": "complete",
          },
          "productRecommendations": Object {
            "data": Array [],
            "status": "complete",
          },
          "provisionedProducts": Object {
            "data": Object {},
            "status": "complete",
          },
          "recentContainers": Object {
            "data": Object {
              "data": Array [],
            },
            "status": "complete",
          },
          "userSiteData": Object {
            "data": Object {
              "currentSite": Object {
                "products": Array [],
                "url": "",
              },
            },
            "status": "complete",
          },
        }
      }
    />
    <RecentSection
      appearance="drawer"
      recentLinks={Array []}
    />
    <CustomLinksSection
      appearance="drawer"
      customLinks={Array []}
    />
  </SwitcherWrapper>
</NavigationAnalyticsContext>
`;
